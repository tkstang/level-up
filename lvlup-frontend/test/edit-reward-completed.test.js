import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import EditRewardCompleted from '../src/components/Admin/rewards/edit-reward/edit-reward-completed';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('EditRewardCompleted should render', () => {
  const store = mockStore({ lvlupApp });
  const edit = shallow(
    <EditRewardCompleted editedReward={{ data: { name: 'Pizza Party', point_cost: 75, description: 'Points are per person' } }} store={store} />,
  );
  expect(shallowToJson(edit)).toMatchSnapshot();
});
