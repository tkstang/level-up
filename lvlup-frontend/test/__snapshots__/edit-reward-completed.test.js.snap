exports[`test EditRewardCompleted should render 1`] = `
<Container
  className="lvl-table">
  <Table
    as="table"
    celled={true}
    color="orange"
    selectable={true}>
    <TableHeader
      as="thead">
      <TableRow
        as="tr"
        cellAs="td">
        <TableHeaderCell
          as="th"
          className="completed-thead"
          textAlign="center">
          Reward Edit Successful!
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody">
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Reward
          </Label>
          Pizza Party
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Point Cost
          </Label>
          75
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Description
          </Label>
          Points are per person
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
  <Grid
    centered={true}>
    <Link
      replace={false}
      to="/admin/rewards">
      <Button
        as="button"
        basic={true}
        color="orange"
        id="completed-btn">
        Back to Rewards
      </Button>
    </Link>
  </Grid>
</Container>
`;
